source/tests/common/test_common.py

```
@@ -24,8 +30,47 @@ def test(self) -> None:
         self.assertEqual(get_xp_precision(xp, "double"), xp.float64)
         self.assertEqual(get_xp_precision(xp, "global"), GLOBAL_NP_FLOAT_PRECISION)
         self.assertEqual(get_xp_precision(xp, "default"), GLOBAL_NP_FLOAT_PRECISION)
+        # NumPy array namespace does not expose bfloat16 – falls back to ml_dtypes
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)